pkg/auth/auth.go

```diff
@@ -84,9 +85,9 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		}
 		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
 	} else {
-		server = getRegistryName(args[0])
+		server = trimScheme(args[0])
```

Comment:
I’d also find it cleaner to centralize parsing, to have a _single_ place that turns user input into (hostPort, optional namespace), or whatever the data is, instead of having `domain(…)` and `splitPath` spread all over the code.

Reply:
Done in `getCredentials`